.github/workflows/pledge-signer-sync/pledge-sync.js

```
@@ -0,0 +1,142 @@
+  where,
+} from "firebase/firestore"
+import fetch from "node-fetch"
+
+const { GALXE_ACCESS_TOKEN, FIRESTORE_USER, FIRESTORE_PASSWORD } = process.env
+
+if (!GALXE_ACCESS_TOKEN || !FIRESTORE_USER || !FIRESTORE_PASSWORD) {
+  console.error("Missing credentials")
+  process.exit(1)
+}
+
+// Limit sync range to last 4 days ( 2 days from last sync + 2 days from now )
+const TARGET_DATE = new Date(Date.now() - 4 * 24 * 60 * 60_000)
+
+const wait = (ms) => new Promise((r) => setTimeout(r, ms))
+
+const getAddresses = async () => {
+  const app = initializeApp({
+    apiKey: "AIzaSyAa78OwfLesUAW8hdzbhUkc5U8LSVH3y7s",
+    authDomain: "tally-prd.firebaseapp.com",
+    projectId: "tally-prd",
+    storageBucket: "tally-prd.appspot.com",
+    messagingSenderId: "567502050788",
+    appId: "1:567502050788:web:bb953a931a98e396d363f1",
+  })
+
+  await signInWithEmailAndPassword(
+    getAuth(app),
+    FIRESTORE_USER,
+    FIRESTORE_PASSWORD
+  )
+
+  const db = getFirestore(app)
+  const dbCollection = collection(db, "address")
+
+  const CHUNK_SIZE = 5_000
+
+  const fetchNextBatch = async (offset) => {
+    let currentQuery = query(
+      dbCollection,
+      orderBy("signedManifesto.timestamp", "desc"),
+      limit(CHUNK_SIZE),
+      where("signedManifesto.timestamp", ">=", TARGET_DATE)
+    )
+
+    if (offset) {
+      currentQuery = query(currentQuery, startAfter(offset))
+    }
+
+    return getDocs(currentQuery).then((snapshot) => snapshot.docs)
+  }
+
+  const allDocs = []
+
+  let offsetDoc
+  let nextBatch = []
+
+  do {
+    nextBatch = await fetchNextBatch(offsetDoc)
+
+    offsetDoc = nextBatch[nextBatch.length - 1]
+
+    allDocs.push(...nextBatch)
+
+    console.log("Retrieved:", allDocs.length, "addresses")
+
+    await wait(1500)
+  } while (nextBatch.length === CHUNK_SIZE)
+
+  return allDocs.map((doc) => doc.id)
+}
+
+const syncGalxe = async () => {
+  const CHUNK_SIZE = 3_000
+
+  const addresses = await getAddresses().catch(() => null)
```

We should let this throw if we fail to get the address, no?

---

Yes, I caught the error here in favor of throwing a custom error a few lines below. Should I log the original error too?

---

Yeah, I feel like debugging this is going to be infuriating enough even with the error when something goes wrong here 😁